### PR TITLE
feat(WC-establish-connection): Fine-tune dApps button

### LINKED ISSUE #14672: [UI - dapps Connection] Add dApp menu button on top of the main wallet layout header with the needed particularities
### Description

[Design](https://www.figma.com/file/HrmZp1y4S77QJezRFRl6ku/Descoped-dApp-Interactions?type=design&node-id=28%3A28851&mode=design&t=LpX0Wx80R6SXcYAl-1)

Details of the `User Story` and scenarios to cover [HERE](https://www.notion.so/Establishing-dApp-connection-934645fe8d704cd1af8e4fc078805377)

  - [x] New button
  - [ ] Tooltip when hovering
Screenshot 2024-05-09 at 17 47 47
  - [x] Floating menu (containing list and placeholder box (we already have SQ component for this type of placeholder). The list shall popuplate the dApps connected model. 
 **NOTE**: Details of the menu will be covered by other task.
Screenshot 2024-05-09 at 17 47 56
- [ ] Add badge that indicates there are some dApps already connected
Screenshot 2024-05-09 at 20 14 42

### PR DESCRIPTION
**Pull request by alexjba**
### What does the PR do

Closes #14672 

Changes:
1. Transform the dApps button into a combo box and add badge indicator + tooltip
2. Add storybook page
3. Adding tests
4. Minor fixes - dApps popup width, ModelUtils imports

### Affected areas

Dapps workflow

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/47811206/f0cd6a99-8b48-4903-ba48-81b76d334644

- [x] I've checked the design and this PR matches it

### CODE REVIEW

**Review by stefandunca — APPROVED** [review]
Good job improving on the architecture also

File: ui/StatusQ/src/StatusQ/Core/Utils/ModelUtils.qml [hunk 1]
```diff
@@ -2,6 +2,7 @@ pragma Singleton
 
 import QtQuick 2.14
 
+import StatusQ 0.1 as StatusQ
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Core/Utils/ModelUtils.qml:5]: Hmm... what is the goal here? :thinking:

File: ui/app/AppLayouts/Wallet/controls/DappsComboBox.qml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    StatusToolTip {
+        id: tooltip
+        objectName: "dappTooltip"
+        visible: root.hovered && !root.down
+        text: qsTr("dApp connections")
+        orientation: StatusToolTip.Orientation.Bottom
+        y: root.height + 14
+    }
+
+    component DAppDelegate: Item {
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/controls/DappsComboBox.qml:87]: Not a big deal, but the ComboBox docu recommends that the `delegate` be an instance of `AbstractButton`; that way you wouldn't have to manually close the popup above. It also helps with tracking the `currentIndex`

See https://doc.qt.io/qt-5/qml-qtquick-controls2-combobox.html#delegate-prop

**Review by caybro — APPROVED** [review]
LGTM

Just some minor nits

File: ui/app/AppLayouts/Wallet/controls/DappsComboBox.qml (hunk 2 above)
Comment (alexjba) [reply ui/app/AppLayouts/Wallet/controls/DappsComboBox.qml:87]: That's true! Will be handled in a different task where we'll focus on the popup itself

File: ui/StatusQ/src/StatusQ/Core/Utils/ModelUtils.qml (hunk 1 above)
Comment (alexjba) [reply ui/StatusQ/src/StatusQ/Core/Utils/ModelUtils.qml:5]: The tests were failing to import StatusQ.Internal and StatusQ was needed. I don't remember why I've added the alias though. Will try it out without it

Comment (caybro) [reply ui/StatusQ/src/StatusQ/Core/Utils/ModelUtils.qml:5]: Ah right, I thought I'd seen this issue before; just wanted to make sure this fixes it :)

**Comment by alexjba** [comment]
@caybro Can't get this test to pass. Seems to fail on master as well. Do you have some time to have a look on it?
Or maybe we can disable it and add a new task?

```
[2024-07-01T07:36:16.031Z] 3: QWARN  : QmlTests::SwapInputPanel::test_setTokenKeyAndAmounts(abc) file:///home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15258/ui/StatusQ/src/StatusQ/Core/Utils/big.min.mjs:7: Error: [big.js] Invalid number (exception occurred during delayed function evaluation)
[2024-07-01T07:36:22.287Z] 3: XFAIL  : QmlTests::SwapInputPanel::test_setTokenKeyAndAmounts(abc) Invalid data expected to fail: abc
[2024-07-01T07:36:22.288Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15258/storybook/qmlTests/tests/tst_SwapInputPanel.qml(142)]
[2024-07-01T07:36:22.288Z] 3: PASS   : QmlTests::SwapInputPanel::test_setTokenKeyAndAmounts(abc)

```

**Comment by caybro** [comment]
> @caybro Can't get this test to pass. Seems to fail on master as well. Do you have some time to have a look on it? Or maybe we can disable it and add a new task?
> 
> ```
> [2024-07-01T07:36:16.031Z] 3: QWARN  : QmlTests::SwapInputPanel::test_setTokenKeyAndAmounts(abc) file:///home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15258/ui/StatusQ/src/StatusQ/Core/Utils/big.min.mjs:7: Error: [big.js] Invalid number (exception occurred during delayed function evaluation)
> [2024-07-01T07:36:22.287Z] 3: XFAIL  : QmlTests::SwapInputPanel::test_setTokenKeyAndAmounts(abc) Invalid data expected to fail: abc
> [2024-07-01T07:36:22.288Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15258/storybook/qmlTests/tests/tst_SwapInputPanel.qml(142)]
> [2024-07-01T07:36:22.288Z] 3: PASS   : QmlTests::SwapInputPanel::test_setTokenKeyAndAmounts(abc)
> ```

XFAIL is a PASS... there must be another (real) issue

**Comment by alexjba** [comment]
> > @caybro Can't get this test to pass. Seems to fail on master as well. Do you have some time to have a look on it? Or maybe we can disable it and add a new task?
> > ```
> > [2024-07-01T07:36:16.031Z] 3: QWARN  : QmlTests::SwapInputPanel::test_setTokenKeyAndAmounts(abc) file:///home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15258/ui/StatusQ/src/StatusQ/Core/Utils/big.min.mjs:7: Error: [big.js] Invalid number (exception occurred during delayed function evaluation)
> > [2024-07-01T07:36:22.287Z] 3: XFAIL  : QmlTests::SwapInputPanel::test_setTokenKeyAndAmounts(abc) Invalid data expected to fail: abc
> > [2024-07-01T07:36:22.288Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15258/storybook/qmlTests/tests/tst_SwapInputPanel.qml(142)]
> > [2024-07-01T07:36:22.288Z] 3: PASS   : QmlTests::SwapInputPanel::test_setTokenKeyAndAmounts(abc)
> > ```
> 
> XFAIL is a PASS... there must be another (real) issue

Ohh, right! Was just searching for the first FAIL match

**Comment by caybro** [comment]
Here's the real culprit:
```
[2024-07-01T12:24:10.613Z] 3: FAIL!  : QmlTests::SwapInputPanel::test_input_greater_than_max_balance() Compared values are not the same
[2024-07-01T12:24:10.613Z] 3:    Actual   (): 0
[2024-07-01T12:24:10.613Z] 3:    Expected (): 5.42
[2024-07-01T12:24:10.613Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15405/storybook/qmlTests/tests/tst_SwapInputPanel.qml(410)]
```

**Comment by alexjba** [comment]
> Here's the real culprit:
> 
> ```
> [2024-07-01T12:24:10.613Z] 3: FAIL!  : QmlTests::SwapInputPanel::test_input_greater_than_max_balance() Compared values are not the same
> [2024-07-01T12:24:10.613Z] 3:    Actual   (): 0
> [2024-07-01T12:24:10.613Z] 3:    Expected (): 5.42
> [2024-07-01T12:24:10.613Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15405/storybook/qmlTests/tests/tst_SwapInputPanel.qml(410)]
> ```

It's also my tests..the ComboBox.hovered is not working when the test runs with `-platform offscreen`

**Comment by caybro** [comment]
@alexjba can you try with this diff (I remember changing that in my PR): https://github.com/status-im/status-desktop/pull/15336/files#diff-b1211fe55c926fb3e8806fad95a9c46175f05adf5e375b6d3ee128b2269f2788R88-R89

**Pull request merged**